scripts/objects/spaceShip.js

```diff
@@ -141,16 +146,33 @@ Game.objects.SpaceShip = function (spec) {
                 mostSafe = possibleLocations[d];
             }
         }
-        spec.center.x = mostSafe.xPos;
-        spec.center.y = mostSafe.yPos;
-        xSpeed = 0;
-        ySpeed = 0;
+        if(mostSafe.xPos && mostSafe.yPos) {
+            spec.center.x = mostSafe.xPos;
+            spec.center.y = mostSafe.yPos;
+            console.log('Space ship at ' + spec.center.x + ': ' + spec.center.y); 
+            xSpeed = 0;
+            ySpeed = 0;
+            safetyFactor = 10; 
+        }
+        else {
+            console.log('ERROR: undefined ship location. Running hyperspace again'); 
+            safetyFactor--; 
+            if(safetyFactor > 2) {
+                hyperspace(allObjectsToAvoid); 
+            } else {
+                console.log('ERROR: No safe locations for hyperspace'); 
+                spec.center.x = undefined;  
+                spec.center.y =  undefined;
+                
+                throw 'hyperspace error'; 

```

Comment:
Controlling logic with a try catch is not ideal, but it is an error case that can be handled in a good manner.